### PR TITLE
fix: Fix errors in Option class

### PR DESCRIPTION
**Pull request by LukasRH**
Cant have generic and not have generic at the same time

### CODE REVIEW

File: Libraries/RhythmGamesBot.Utils/Option.cs [hunk 1]
```diff
@@ -51,11 +51,11 @@ public Option<T> Replace(T value)
                 T temp = this._value;
                 this._value = value;
 
-                return temp;
+                return new Option<T>(temp);
```
Comment (cgytrus) [inline Libraries/RhythmGamesBot.Utils/Option.cs:54]: shouldn't it just return `this`?

Comment (LukasRH) [reply Libraries/RhythmGamesBot.Utils/Option.cs:54]: According to the comment, it should return the old value, at least that it how I understand it. 
But my initial though was the same as yours

Comment (cgytrus) [reply Libraries/RhythmGamesBot.Utils/Option.cs:54]: oh wait, I didn't notice `this._value = value;`

Comment (LukasRH) [reply Libraries/RhythmGamesBot.Utils/Option.cs:54]: yeah, it replaces the value and returns the old

Comment (cgytrus) [reply Libraries/RhythmGamesBot.Utils/Option.cs:54]: then yeah, it makes sense

**Review by cgytrus — APPROVED** [review]

**Pull request merged**